pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -181,183 +186,183 @@ public void stop() {
     LOGGER.info("Wait completed: Number of completing segments = {}", _numCompletingSegments.get());
   }
 
-  protected boolean isLeader(String tableName) {
-    return _leadControllerManager.isLeaderForTable(tableName);
-  }
-
-  protected boolean isConnected() {
-    return _helixManager.isConnected();
-  }
-
   /**
-   *
-   * @param tableConfig
-   * @param emptyIdealState may contain HLC segments if both HLC and LLC are configured
+   * Sets up the initial segments for a new LLC real-time table.
+   * <p>NOTE: the passed in IdealState may contain HLC segments if both HLC and LLC are configured.
    */
-  public void setupNewTable(TableConfig tableConfig, IdealState emptyIdealState)
-      throws InvalidConfigException {
-    final StreamConfig streamConfig = new StreamConfig(
-        tableConfig.getTableName(),
-        tableConfig.getIndexingConfig().getStreamConfigs());
-    int partitionCount = getPartitionCount(streamConfig);
-    List<String> currentSegments = getExistingSegments(tableConfig.getTableName());
-    // Make sure that there are no low-level segments existing.
-    if (currentSegments != null) {
-      for (String segment : currentSegments) {
-        if (!SegmentName.isHighLevelConsumerSegmentName(segment)) {
-          // For now, we don't support re-creating the low-level realtime segments
-          throw new RuntimeException("Low-level segments already exist for table " + tableConfig.getTableType());
-        }
-      }
+  public void setUpNewTable(TableConfig tableConfig, IdealState idealState) {
+    Preconditions.checkState(!_isStopping, "Segment manager is stopping");
+
+    String realtimeTableName = tableConfig.getTableName();
+    LOGGER.info("Setting up new LLC table: {}", realtimeTableName);
+
+    // Make sure all the existing segments are HLC segments
+    List<String> currentSegments = getAllSegments(realtimeTableName);
+    for (String segmentName : currentSegments) {
+      Preconditions.checkState(SegmentName.isHighLevelConsumerSegmentName(segmentName),
```

Comment:
Does this show up correctly in the table create command line on the client side?

Reply:
Yes, both (IllegalStateException and RuntimeException) are returned as INTERNAL_SERVER_ERROR

Reply:
Added the TODO. This PR keeps all the old API behaviors the same.